### PR TITLE
Restructured config

### PR DESCRIPTION
**Pull request by theory-in-progress**
- Restructured config so that arguments are defined in cofing.yaml

### CODE REVIEW

**Comment by vfdev-5** [comment]
@theory-in-progress CI is broken any ideas why ?

File: src/tests/ci-configs/vision-segmentation-simple.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+seed: 666
+data_path: ~/data
+train_batch_size: 2
+eval_batch_size: 2
+num_workers: 2
+max_epochs: 2
+train_epoch_length: 2
+eval_epoch_length: 2
```
Comment (vfdev-5) [inline src/tests/ci-configs/vision-segmentation-simple.yaml:8]:
```suggestion
train_epoch_length: 4
eval_epoch_length: 4
```
as original cmd line options

File: src/tests/ci-configs/text-classification-launch.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+seed: 666
+backend: gloo
```
Comment (vfdev-5) [inline src/tests/ci-configs/text-classification-launch.yaml:2]: What backend does here ? It should be passed as cmd arg

File: scripts/run_tests.sh [hunk 3]
```diff
@@ -49,15 +37,7 @@ run_launch() {
...
-    torchrun \
-      --nproc_per_node 2 \
-      main.py --backend gloo --data_path ~/data \
-      --train_batch_size 2 \
-      --eval_batch_size 2 \
-      --num_workers 1 \
-      --max_epochs 2 \
-      --train_epoch_length 4 \
-      --eval_epoch_length 4
+    torchrun main.py ../../src/tests/ci-configs/$1-launch.yaml
```
Comment (vfdev-5) [inline scripts/run_tests.sh:40]: This command is not the same as the old one:
```suggestion
    torchrun --nproc_per_node 2 main.py ../../src/tests/ci-configs/$1-launch.yaml --backend gloo
```

File: src/tests/ci-configs/text-classification-launch.yaml (hunk 2 above)
Comment (theory-in-progress) [reply src/tests/ci-configs/text-classification-launch.yaml:2]: Yeah, actually resolving that right now

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**